src/math/crypto/poseidon/poseidon.zig

```diff
@@ -0,0 +1,175 @@
+const std = @import("std");
```

Comment:
can we include a comment with the porting lineage? i.e. we are porting from `stark-rs`, which in turn ported from `pathfinder`

something like: https://github.com/xJonathanLEI/starknet-rs/blob/0857bd6cd3bd34cbb06708f0a185757044171d8d/starknet-crypto/src/poseidon_hash.rs#L2